lib/Targets/AIEVecToCpp/TranslateAIEVecToCpp.cpp

```diff
@@ -2438,6 +2490,44 @@ bool CppEmitter::hasBlockLabel(Block &block) {
   return blockMapper.count(&block);
 }
 
+template <typename ElTy>
+static bool hasSameDenseValue(DenseIntElementsAttr dense,
+                              std::string &firstValue) {
+  SmallVector<ElTy> denseValues;
+  for (APInt elem : dense) {
+    denseValues.push_back(elem.getLimitedValue());
+  }
+  ElTy firstV = denseValues[0];
+  if (llvm::all_of(denseValues, [firstV](ElTy v) { return v == firstV; })) {
+    firstValue = std::to_string(firstV);
+    return true;
+  }
+  return false;
+}
+
+template <typename ElTy>
+static bool hasSameDenseValue(DenseFPElementsAttr dense,
+                              std::string &firstValue) {
+  SmallVector<ElTy> denseValues;
+  for (APFloat elem : dense) {
+    denseValues.push_back(elem.convertToFloat());
+  }
+  ElTy firstV = denseValues[0];
+
+  if (llvm::all_of(denseValues, [firstV](ElTy v) { return v == firstV; })) {
+    firstValue = std::to_string(firstV);
+
+    if (firstValue == "inf") {
+      firstValue = "3.40282347e+38F";
```

Comment:
I am curious about this magical value. Is this something that could come from https://en.cppreference.com/w/cpp/types/numeric_limits or from LLVM target description?

Reply:
This change set is not ready for review, Ronan. I haven't requested anyone for review yet. I thought no one can see this. Once I finish it, I will add you for review.

Reply:
Ah yes, this is actually a draft. Sorry for dropping by. :-)